actors/builtin/miner/miner_state.go

```diff
@@ -970,6 +975,81 @@ func (st *State) MeetsInitialPledgeCondition(balance abi.TokenAmount) bool {
 	return available.GreaterThanEqual(st.InitialPledgeRequirement)
 }
 
+// pre-commit expiry
+func (st *State) AddPreCommitExpiry(store adt.Store, currEpoch abi.ChainEpoch, sealProof abi.RegisteredSealProof, sectorNum abi.SectorNumber) error {
+	msd, ok := MaxSealDuration[sealProof]
+	if !ok {
+		return xerrors.Errorf("no max seal duration set for proof type: %d", sealProof)
+	}
+	// The +1 here is critical for the batch verification of proofs. Without it, if a proof arrived exactly on the
+	// due epoch, ProveCommitSector would accept it, then the expiry event would remove it, and then
+	// ConfirmSectorProofsValid would fail to find it.
+	expiryBound := currEpoch + msd + 1
+
+	// create BitField for this sector
+	bf := abi.NewBitField()
+	bf.Set(uint64(sectorNum))
+
+	// Load BitField Queue for sector expiry
+	quant := NewQuantSpec(WPoStChallengeWindow, st.ProvingPeriodStart)
```

Comment:
We create this twice, so extract `st.QuantSpecEveryDeadline()`.

Reply:
This is done.